src/fabric_cicd/fabric_workspace.py

```diff
@@ -431,6 +432,27 @@ def _publish_item(
                 logger.info(f"Skipping publishing of {item_type} '{item_name}' due to exclusion regex.")
                 return
 
+        # Skip publishing if the item is not in the include list
+        if self.items_to_include is not None:
+            current_item = f"{item_name}.{item_type}"
+            logger.debug(f"Checking if '{current_item}' is in items_to_include: {self.items_to_include}")
+
+            # Check for exact match or case-insensitive match
+            # Normalize include list to a lowercase set for efficient lookups
+            normalized_include_set = {include_item.lower() for include_item in self.items_to_include}
+
+            # Check for exact match or case-insensitive match
+            match_found = current_item in self.items_to_include or current_item.lower() in normalized_include_set
+
+            if not match_found:
+                item.skip_publish = True
+                logger.info(f"Skipping publishing {item_name}.{item_type} as it is not in the items to include list.")
+                logger.debug(f"Include list: {self.items_to_include}")
+                return
+            logger.info(f"Publishing {item_name}.{item_type} as it is in the items to include list.")
+        else:
+            logger.info(f"Publishing {item_name}.{item_type} as no items to include list is provided.")
```

Comment:
Lines 452 - 454 can be removed or change the logger type to debug so the console is not cluttered.